src/app/shared/components/common/directives/collapse-on-scroll.directive.ts

```diff
@@ -0,0 +1,92 @@
+import { Directive, ElementRef, Input, OnDestroy, OnInit, Renderer2 } from "@angular/core";
+import { DomController, ScrollDetail } from "@ionic/angular";
+import { Subscription } from "rxjs/internal/Subscription";
+import { Subject, filter, map } from "rxjs";
+import { AppConfigService } from "src/app/shared/services/app-config/app-config.service";
+import { IAppConfig } from "src/app/shared/model";
+
+/** The id of the ion-content component upon which scrolling will trigger the collapse */
+const SCROLL_AREA_ID = "collapseOnScrollTargetScrollArea";
+
+@Directive({
+  selector: "[plhCollapseOnScroll]",
+})
+export class CollapseOnScrollDirective implements OnInit, OnDestroy {
+  @Input("plhCollapseOnScroll") scrollEvents$: Subject<any>;
+
+  private hidden: boolean = false;
+  private triggerDistance: number = 20;
+  scrollSubscription$: Subscription;
+  defaultHeight: number;
+  collapsing: boolean;
+
+  constructor(
+    private element: ElementRef,
+    private renderer: Renderer2,
+    private domCtrl: DomController,
+    private appConfigService: AppConfigService
+  ) {}
+
+  ngOnInit() {
+    this.subscribeToAppConfigChanges();
+    if (this.collapsing) {
```

Comment:
nit(blocking) - could you move all this code into a separate function with a description of what it does?